share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,809 @@
+```
+
+* Below are the set of examples which shows how to extract specific characters from a string
+```
+def first s = s | explode | take 1| cat
+def second s = s | explode | tail | take 1 | cat
+def third s = s | explode| tail | tail | take 1 |cat
+def fourth s =  s | explode | tail | tail | tail | take 1| cat
+def last s = s| explode | reverse |take 1 | cat
+```
+
+* Try out these different set of examples and capture the result
+```
+def roll s = "{fourth s}{first s}{second s}{third s}"
+def exch s = "{second s}{first s}{third s}{fourth s}"
+```
+
+* Try out these examples and record the results
+```
+def what fullstring = fullstring| roll | roll | roll| exch |  roll
+def fb fullstring = fullstring | roll |roll | roll
+def fc fullstring = fullstring | roll | exch |fb
+def fd fullstring = fullstring | exch | fc | exch
+```
+
+* This examples convert the string "seat" -> "eats"
+```
+wake -x 'cat((tail (explode "seat")) ++(take 1(explode "seat")))'
+```
+
+### Tuple
+A tuple is collection of items of different types. `get`,`set` and `edit` are the three inbuilt methods that can be used with the tuples.
+* Below example shows the declaration of a tuple. 
+```
+tuple Collection =
+  global IntValue : Integer
+  global DoubleValue : Double
+  global StringValue : String
+```
+
+`set<tuple name><field Name>` 
+`set` operation sets(writes) value to the tuple field
+`get<tuple name><field Name>` 
+`get` operation gets(reads) value to the tuple field
+`edit<tuple name><field Name>` 
+`edit` operation moidfies value to the tuple field
+
+### Map
+Map is a higher order function which takes a function and applies the same to every element in the list.
+
+* Below example shows how to declare a list of strings and apply map function to it. Map is a predefined function that maps to the every element in the list
+```
+def words = "ache", "vile", "amid", "evil", "ogre" , Nil
+def mapping words = words | map roll
+def mapping2 words = map exch words
+```
+
+### Higher Order function
+A function which does one of the following is a higher order function
+1. Takes one or more functions as argument
+2. Returns a function as its result
+
+* Funtion tea shows the way to pass a function as a argument 
+```
+def tea f = f 4
+```
+* `Map is the best example for a Higher Order Function` 
+
+* Try out these examples and record the result
+```
+def op1 = triple (inc 1)
+def op2 = inc 1 | triple
+def op3 = double (inc (triple 1))
+def op4 = triple (inc (double 1))
+def op5 = tea double
+def op6 = tea inc
+def op7 = tea inc | double 
+def op8 = map double (1,2,3,Nil)
+def op9 = map double (1,2,3,Nil) | map inc
+```
+* Below example multiplies a given number by 6. The other way to implement this is `def times6 x = double x | triple`
+```
+def times6 x = triple (double x)
+```
+
+### Pattern Matching
+The first thing that comes to mind is string matching. It is similar to the Switch statement where it will check for the matching case and outputs appropriate results.
```

Comment:
You have three examples of simple string matching. Is that it? Is regular expression matching thing? Or are we going to show more examples later... reading on.